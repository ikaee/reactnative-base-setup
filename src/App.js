/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @format
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import React from 'react'
import { Text, View } from 'react-native'
import SampleComponent from './components/SampleComponent'
import constant from './variables/string'
import styles from './app.style'

const App = () => (
  <View style={styles.container}>
    <SampleComponent onPress={() => {}} label={constant.sampleButtonText} />
    <Text style={styles.welcome}>{constant.sampleText}</Text>
  </View>
)

export default App
