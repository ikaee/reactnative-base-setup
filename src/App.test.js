// import { mount } from 'enzyme'
import { shallow } from 'enzyme'
import React from 'react'
import App from './App'

describe('<App/>', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<App />)
  })

  it('shows a SampleComponent', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
