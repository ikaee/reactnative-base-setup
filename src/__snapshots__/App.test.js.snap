// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> shows a SampleComponent 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCF0",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <SampleComponent
    label="Click me"
    onPress={[Function]}
  />
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to React Native!
  </Text>
</View>
`;
